pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller.go

```diff
@@ -77,16 +81,88 @@ func NewClusterMemberRemovalController(
 			masterNodeInformer,
 			masterMachineInformer,
 			kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().ConfigMaps().Informer(),
+			kubeInformersForNamespaces.InformersFor("kube-system").Core().V1().ConfigMaps().Informer(),
 		).ToController("ClusterMemberRemovalController", eventRecorder.WithComponentSuffix("cluster-member-removal-controller"))
 }
 
 func (c *clusterMemberRemovalController) sync(ctx context.Context, _ factory.SyncContext) error {
-	// stop if the machine API is not functional
+	var errs []error
+
+	if err := c.removeMemberWithoutMachine(ctx); err != nil {
+		errs = append(errs, err)
+	}
+
+	// only attempt to scale down if the machine API is functional
 	if isFunctional, err := c.machineAPIChecker.IsFunctional(); err != nil {
-		return err
+		errs = append(errs, err)
+		return kerrors.NewAggregate(errs)
 	} else if !isFunctional {
 		return nil
 	}
+
+	if err := c.attemptToScaleDown(ctx); err != nil {
+		errs = append(errs, err)
+	}
+
+	return kerrors.NewAggregate(errs)
+}
+
+// attemptToScaleDown attempts to remove the member only once we have identified that
+// a Machine resource is being deleted and a replacement member has been created
+func (c *clusterMemberRemovalController) attemptToScaleDown(ctx context.Context) error {
+	masterMachines, err := c.masterMachineLister.List(c.masterMachineSelector)
+	if err != nil {
+		return err
+	}
+	memberMachines := ceohelpers.FilterMachinesWithMachineDeletionHook(masterMachines)
+
+	// machines with the deletion hooks are the ones that host etcd members
+	// if the number of master machines is > controlPlane.replicas based on data from the cache
+	desiredControlPlaneReplicasCount, err := ceohelpers.ReadDesiredControlPlaneReplicasCount(c.configMapListerForKubeSystemNamespace)
+	if err != nil {
+		return err
+	}
+	if len(memberMachines) <= desiredControlPlaneReplicasCount {
+		klog.V(4).Infof("haven't found a replacement machine, the number of desired control plane replicas: %d must be greater than the current number of machines that host an etcd member: %d", desiredControlPlaneReplicasCount, len(memberMachines))
+		return nil
+	}
```

Comment:
> the number of desired control plane replicas: %d must be greater than the current number of machines that host an etcd member

It's actually the opposite since we want an excess of voting member etcd machines or greater than the desired control-plane size, in order to select a replacement member to scale-down.
So we can probably make this log statement more accurate e.g:
```suggestion
	if len(memberMachines) <= desiredControlPlaneReplicasCount {
		klog.V(4).Infof("Ignoring scale-down since the number of etcd voting member machines (%d) < desired number of control-plane replicas (%d) ", len(memberMachines), desiredControlPlaneReplicasCount)
		return nil
	}
```